python/tvm/topi/adreno/conv2d_transpose_nchw.py

```diff
@@ -0,0 +1,383 @@
+            )
+    else:
+        batch, in_channel_chunks, in_height, in_width, in_channel_block = Input.shape
+
+    if len(Filter.shape) == 4:
+        in_filter_channels, out_channels, kernel_h, kernel_w = Filter.shape
+        out_channel_chunks, out_channel_block, out_channel_tail = split_to_chunks(out_channels, 4)
+
+        if autotvm.GLOBAL_SCOPE.in_tuning:
+            kshape = (in_filter_channels, out_channel_chunks, kernel_h, kernel_w, out_channel_block)
+            Filter = tvm.te.placeholder(kshape, Filter.dtype, name="kernel_placeholder")
+        else:
+            convert_from4d = True
+            Filter = pack_filter(
+                Filter,
+                "IOHW",
+                out_channel_chunks,
+                out_channel_block,
+                out_channel_tail,
+                in_filter_channels,
+                in_channel_chunks,
+                in_channel_block,
+                in_channel_tail,
+                kernel_h,
+                kernel_w,
+            )
+    else:
+        in_filter_channels, out_channel_chunks, kernel_h, kernel_w, out_channel_block = Filter.shape
+
+    cfg.stride = stride
+
+    pad_top, pad_left, pad_bottom, pad_right = nn.get_pad_tuple(padding, (kernel_h, kernel_w))
+
+    out_width_orig = out_width = (
+        (in_width - 1) * stride_w + kernel_w - pad_left - pad_right + outpad_width
+    )
+    pad_left = kernel_w - 1 - pad_left
+    pad_right = kernel_w - 1 - pad_right + outpad_width
+    dilated_width = stride_w * (in_width - 1) + 1
+
+    out_height_orig = out_height = (
+        (in_height - 1) * stride_h + kernel_h - pad_top - pad_bottom + outpad_height
+    )
+    pad_top = kernel_h - 1 - pad_top
+    pad_bottom = kernel_h - 1 - pad_bottom + outpad_height
+    dilated_height = stride_h * (in_height - 1) + 1
+
+    if out_height % 2 != 0:
+        out_height += 1
+    if out_width % 2 != 0:
+        out_width += 1
+
+    if out_height % 4 != 0:
+        out_height += 2
+    if out_width % 4 != 0:
+        out_width += 2
+
+    # compute pad
+    temp = te.compute(
+        (
+            batch,
+            in_channel_chunks,
+            pad_top + dilated_height + pad_bottom,
+            pad_left + dilated_width + pad_right,
+            in_channel_block,
+        ),
+        lambda n, c, y, x, cb: tvm.tir.if_then_else(
+            tvm.tir.all(
+                x >= pad_left,
+                x < pad_left + dilated_width,
+                tvm.tir.indexmod(x - pad_left, stride_w).equal(0),
+                y >= pad_top,
+                y < pad_top + dilated_height,
+                tvm.tir.indexmod(y - pad_top, stride_h).equal(0),
+            ),
+            Input[
+                n,
+                c,
+                tvm.tir.indexdiv(y - pad_top, stride_h),
+                tvm.tir.indexdiv(x - pad_left, stride_w),
+                cb,
+            ],
+            tvm.tir.const(0.0, Input.dtype),
+        ),
+        name="pad_temp",
+    )
```

Comment:
Instead of this code, you can use `add_pad` from `adreno/utils.py` or `nn.pad`.

Reply:
The conv2d_transpose padding algo implementation is different compared to regular explicit padding like conv2d ops. 
So we can't use existing add_pad or nn.pad for this.

Thanks for review.